### DISCUSSION TITLE
TerriaMap deployment for UNL - advice

### DISCUSSION THREAD

**Discussion by AnaBelgun** [TerriaJS/terriajs#6553] [General] [answered]
Ticket to detail responses to Universidad Nacional de Loja (UNL) request for clarifications around: 
1) Language switch
2) Server 
3) 3D data

**Comment by staffordsmith83** [comment, marked as answer]
### To enable the language switcher:

<img width="286" alt="Pasted image 20220921102842" src="https://user-images.githubusercontent.com/7980991/191410906-9742f1cd-e9b3-4c3c-8f27-6c23b82ef64f.png">


The settings for this are in the TerriaMap repo, here...
`/TerriaMap/wwwroot/config.json`
Change this by modifying `enabled` property in config.json, and adding "es" to the languages object:
```json
 "languageConfiguration": {
      "enabled": true,
      "debug": false,
      "languages": {
	    "es": "Espanol",
        "en": "English"
      },
      "fallbackLanguage": "en"
    },
```


### To deploy Terria to a server

Please see https://docs.terria.io/guide/deploying/deploying-to-aws/ . I will update this doc this week, so check next week and should be helpful for you.


### Adding sensor data to Terria
E.g. weather station data.

This is very dependent on what your weather station can provide. If the weather station can make a live updating `csv` of its data available on the internet somewhere, then you can add this easily to Terria as a `csv` catalog item. Please see https://docs.terria.io/guide/connecting-to-data/catalog-type-details/csv/

If the data updates regularly, and you want these updates to show live, then there is the `polling` trait (setting) for these csv items. Please see https://docs.terria.io/guide/connecting-to-data/catalog-type-details/csv/#pollingtraits

If you weather station does not make its data available on the internet, then you will need to do this yourself. If you use AWS they have a lot of IoT services that are designed to filter and process sensor data and make it available online in various ways. This article has some useful information but dont worry if its more complex than your needs, just an introduction: https://docs.aws.amazon.com/iotanalytics/latest/userguide/iotanalytics-ug.pdf

Alternatively, you could write a script to run on your servals at set time intervals. Something very simple like this.
```
curl foo.bar | perl -pi "some expression here" > /var/spool/website/weather.csv
```
Again the steps are - retrieve the data from the weather station, filter the data, and make it available as a csv!

### Updating colours in UI
Please see this link https://docs.terria.io/guide/customizing/skinning/

### How to display 3D Data, e.g. Lidar:
Your data needs to be converte to `3d-tiles` format and hosted somewhere on the internet. If you can get your data into 3d-tiles format you can copy it to an AWS S3 bucket, or a Google Cloud Storage bucket. If you are not familiar with this, then we recommend to use Cesium Ion - they have a service that will convert 3d data in many formats into the correct format and host it for you!

Give this a try: https://cesium.com/platform/cesium-ion/

You would then add the Cesium Resource that is created to your Terria Catalog as a Cesium3DTilesCatalogItem see https://docs.terria.io/guide/connecting-to-data/catalog-type-details/3d-tiles/

Make sure to fill out the ionAssetId and ionAccessToken traits, here is an example from my personal Cesium Ion account:
```json
 {
      "name": "Cesium 3dtiles hosted example",
      "type": "3d-tiles",
      "ionAssetId": 1029466,
      "ionAccessToken": "dfakdhflaksdhfaslkfhasdkfjhasldkfjhsadkfjhsadkfhasdflkjahsdflkasdhflkajsdhflkasdjhflksadjfhalksdjfhsadkfh"
    },
```

It works!
<img width="1114" alt="Screen Shot 2022-09-21 at 11 04 31 am" src="https://user-images.githubusercontent.com/7980991/191410787-80f11eda-eb2b-42bb-ae03-4c285bf8b6f5.png">

**Comment by feralex12131415** [comment]
thanks so much, for all the answers
